x/sequencer/proposal/block_validator.go

```diff
@@ -0,0 +1,90 @@
+package proposal
+
+import (
+	"runtime"
+	"sync"
+
+	"github.com/cometbft/cometbft/libs/log"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/warp-contracts/sequencer/x/sequencer/controller"
+	"github.com/warp-contracts/sequencer/x/sequencer/keeper"
+
+	"github.com/warp-contracts/syncer/src/utils/task"
+)
+
+type Block struct {
+	ctx sdk.Context
+	txs []sdk.Tx
+}
+
+// Validates transactions in the block proposed by the Proposer.
+// Where required, validation is done sequentially.
+// In other cases, validation occurs in separate threads per transaction.
+type BlockValidator struct {
+	*task.Task
+
+	keeper     *keeper.Keeper
+	controller controller.ArweaveBlocksController
+	logger     log.Logger
+	Input      chan *Block
+	Output     chan bool
+}
+
+func newBlockValidator(keeper *keeper.Keeper, controller controller.ArweaveBlocksController, logger log.Logger) *BlockValidator {
+	validator := new(BlockValidator)
+	validator.keeper = keeper
+	validator.controller = controller
+	validator.logger = logger
+	validator.Input = make(chan *Block)
+	validator.Output = make(chan bool)
+
+	validator.Task = task.NewTask(controller.GetConfig(), "block-validator").
+		WithSubtaskFunc(validator.run).
+		WithWorkerPool(runtime.NumCPU(), 1)
+
+	return validator
+}
+
+func (v *BlockValidator) run() error {
+	for block := range v.Input {
```

Comment:
This will prolong closing of sequencer.
All blocks in the Input channel will have to be verified before closing. Either close the Input somewhere or listen on  v.Ctx.Done()